api/revoke/revoke_test.go

```diff
@@ -0,0 +1,123 @@
+package revoke
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"bitbucket.org/liamstask/goose/lib/goose"
+	"github.com/cloudflare/cfssl/api"
+	"github.com/cloudflare/cfssl/certdb"
+)
+
+// Largely duplicates the functionality of prepDB in certdb_test
+func prepDB() (db *sql.DB, err error) {
+	db, err = sql.Open("sqlite3", ":memory:")
+	if err != nil {
+		return nil, err
+	}
+
+	var dbconf *goose.DBConf
+	dbconf, err = goose.NewDBConf("../../certdb/sqlite/", "test", "")
+	if err != nil {
+		return nil, err
+	}
+
+	err = goose.RunMigrationsOnDb(dbconf, "../../certdb/sqlite/", 1, db)
+	if err != nil {
+		return nil, err
+	}
+
+	expirationTime := time.Now().AddDate(1, 0, 0)
+	var cert = &certdb.CertificateRecord{
+		Serial:    "1",
+		CALabel:   "",
+		Status:    "",
+		Reason:    0,
+		ExpiresAt: &expirationTime,
+		RevokedAt: nil,
+		PEM:       "unexpired cert",
+	}
+
+	err = certdb.InsertCertificate(db, cert)
+	if err != nil {
+		return nil, err
+	}
+
+	return
+}
+
+func testRevokeCert(t *testing.T, db *sql.DB, serial string, reason *string) (resp *http.Response, body []byte) {
```

Comment:
just nitpick: I think it is fine to use "reason string", the empty value of string is ""